lib/Analysis/Utility.cpp

```diff
@@ -290,7 +290,7 @@ unsigned ScanLoweringHelper::getNonAxisNumBlocks() {
 bool ScanLoweringHelper::isSupported() {
   // TODO: Support the following cases:
   // 1. Scan on non-blocking encodings
-  if (!isa<BlockedEncodingAttr>(getEncoding()))
+  if (!isa<BlockedEncodingAttr>(srcEncoding))
```

Comment:
Do you plan to make it more general for other encodings?

Reply:
Yes, in the future, but it's not super high prio ATM, as I would expect it not to be blocking like reduction was.